frontend/build.rs

```diff
@@ -0,0 +1,58 @@
+use std::fs::File;
+use std::path::PathBuf;
+///
+/// Build Script
+/// This is run as a pre-build step -- before the rust backend is compiled.
+/// NOTE: Should be included in root's build script
+///
+use std::{io::Write, process::Command};
+
+/*
+ * Note 1: this file was written for *nix systems -- it likely won't
+ * work on windows!
+ */
+
+#[allow(dead_code)]
+fn shell(command: &str) {
+    // println!("build.rs => {}", command);
+
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg(command)
+        .output()
+        .expect(format!("Failed to run {cmd}", cmd = command).as_str());
+
+    // println!("build.rs => {:?}", output.stdout);
+    let mut file = File::create("build.log").expect("Couldn't create file...");
+    file.write(b"build log\n\n\n\nSTDOUT:\n")
+        .expect("Couldn't write to build log");
+    file.write_all(&output.stdout)
+        .expect("Couldn't write to build log");
+    file.write(b"\n\n\n\nSTDERR:\n")
+        .expect("Couldn't write to build log");
+    file.write_all(&output.stderr)
+        .expect("Couldn't write to build log");
+}
+
+pub fn build_types() {
+    let dir = env!("CARGO_MANIFEST_DIR");
+
+    let inputs = vec![PathBuf::from_iter([dir, "backend"])];
+    let output = PathBuf::from_iter([dir, "frontend/src/types/rust.d.ts"]);
+
+    tsync::generate_typescript_defs(inputs, output, false);
+}
+
+pub fn build_frontend() {
+    // Only install frontend dependencies when building release
+    // #[cfg(not(debug_assertions))]
+    shell("cd frontend && npm install --frozen-lockfile");
+
+    // Only build frontend when building a release
+    #[cfg(not(debug_assertions))]
+    shell("cd frontend && npm build");
+}
+
+// HACK: Dummy main to run with `cargo frontend` command
+#[allow(dead_code)]
+fn main() {}
```

Comment:
Looks like `build_frontend` should be called here. Are you planning to do this in a request with frontend initialization?

Reply:
There are multiple limitations with cargo, that I tried to work around: 

1. Cargo can't trigger build-script in subfolders
2. Cargo needs either `main.rs`, `lib.rs`, `[lib]` section or `[[bin]]` section in order to run *anything*. 

First work around worked like this: all functions were defined in `build.rs` file in frontend directory, that would be included in root's `build.rs` file (using [`include`](https://doc.rust-lang.org/std/macro.include.html) macro) and invoked there

Now `frontend` is a separate workspace, which will build itself and will be unrelated to root's build script. In that way it feels more convenient, it seems
With that, `frontend` feature will be removed since it's practically do nothing now. All commands mentioned in [there](https://github.com/qoollo/bob-management/pull/9#discussion_r1327863367) will work just fine.

Reply:
Maybe there is a more elegant way to do this, but I can't wrap my head around it

Reply:
there is also [`rust-script`](https://github.com/fornwall/rust-script), but I couldn't get it to work